pkg/operator/ceph/file/filesystem.go

```diff
@@ -231,18 +231,20 @@ func (f *Filesystem) doFilesystemCreate(context *clusterd.Context, clusterInfo *
 
 	fslist, err := client.ListFilesystems(context, clusterInfo)
 	if err != nil {
-		return errors.Wrap(err, "Unable to list existing filesystem")
+		return errors.Wrap(err, "failed to list existing filesystem(s)")
 	}
-	if len(fslist) > 0 && !client.IsMultiFSEnabled() {
-		return errors.Errorf("cannot create multiple filesystems. enable %s env variable to create more than one", client.MultiFsEnv)
+	// This check prevents from concurrent CephFilesystem CRD trying to create a filesystem
+	// Whoever gets to create the Filesystem first wins the race, then we fail if that cluster is not Ceph Pacific and one Filesystem is present
+	if len(fslist) > 0 && !clusterInfo.CephVersion.IsAtLeastPacific() {
+		return errors.New("multiple filesystems are only supported as of ceph pacific")
```

Comment:
If someone is already using multiple filesystems in experimental mode, this would break their filesystem reconcile until they upgrade to pacific. What if we keep the env var that allows multiple filesystems on pre-Pacific? Then we just don't need to worry about the backward compat issue even if we don't expect anyone to be using it in experimental mode.

Reply:
I feel like it's a good time to deprecate it though... No? :)

Reply:
What about this approach?
- Rook would no longer set the "enable multiple filesystem" ceph flag at all. If pre-pacific users want to enable multiple filesystems, they have to enable it in the toolbox.
- Rook removes all checks for multiple filesystems and we just allow it. But if pre-pacific users haven't set the ceph flag to enable it, rook will fail to configure them.

Reply:
1. I think Rook should handle the upgrade and apply `enable_multiple` since it is the default in Pacific. It's cheap to execute that command anyways and saves the burden for the users to do it
2. Given that we support Nautilus/Octopus/Pacific (soon), it looks risky to me to open up that possibility. This is true for the upcoming 1.6 release

Thoughts? I'm still in favor of keeping things the way they are. I'd be open to adding the `IsMultiFSEnabled` check to preserve pre-pacific multi-fs clusters, although I highly doubt they exist :)

Reply:
> 1. I think Rook should handle the upgrade and apply `enable_multiple` since it is the default in Pacific. It's cheap to execute that command anyways and saves the burden for the users to do it

Yes, sounds good to ensure multiple are enabled in Pacific.

> 2. Given that we support Nautilus/Octopus/Pacific (soon), it looks risky to me to open up that possibility. This is true for the upcoming 1.6 release
> 
> Thoughts? I'm still in favor of keeping things the way they are. I'd be open to adding the `IsMultiFSEnabled` check to preserve pre-pacific multi-fs clusters, although I highly doubt they exist :)

I don't think we can prevent pre-Pacific users from using multiple filesystems. With the current checks, for example it seems Nautilus/Octopus users would be prevented from using multiple filesystems. Is there a way to allow it still that I'm missing?

Reply:
Right now there is none that's why I was proposing to actually keep the operator setting, what do you think? We can remove this flag during the 1.7 cycle.

Reply:
Agreed, we can keep the setting for now. I was misunderstanding the proposal.